src/permission/fs_permission.cc

```diff
@@ -0,0 +1,199 @@
+#include "fs_permission.h"
+#include "base_object-inl.h"
+#include "v8.h"
+
+#include <fcntl.h>
+#include <limits.h>
+#include <stdlib.h>
+#include <algorithm>
+#include <filesystem>
+#include <iostream>
+#include <string>
+#include <vector>
+
+std::string WildcardIfDir(std::string res) noexcept {
```

Comment:
WildcardIfDir and FreeRecursivelyNode are static, so these should be in an anonymous namespace like this:
```suggestion
namespace {

std::string WildcardIfDir(std::string res) noexcept {
```

Reply:
Why is it required? I'll apply the suggestion, but I couldn't find any reference to it in our cpp guideline.

Reply:
Thank you! I think somehow it should be linked to our cpplint.